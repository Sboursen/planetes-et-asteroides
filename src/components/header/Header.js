import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
  const setActiveStyle = (navData) => {
    const result = navData.isActive ? 'underline text-white font-bold' : '';
    return result;
  };
  return (
    <header className="flex flex-row justify-between items-center py-2 px-6 bg-header">
      <NavLink
        className="h-full flex flex-row gap-3 justify-between items-center text-4xl"
        to="/"
      >
        Planets and asteroids
      </NavLink>
      <nav className="flex flex-row gap-3 justify-between items-center">
        <NavLink className={setActiveStyle} to="/">
          Home
        </NavLink>
        <NavLink className={setActiveStyle} to="/about">
          About
        </NavLink>
        <div className="w-2 h-4 border-r-2 border-white" />
        <NavLink className={setActiveStyle} to="/details">
          Sort
        </NavLink>
      </nav>
    </header>
  );
};
export default Header;
