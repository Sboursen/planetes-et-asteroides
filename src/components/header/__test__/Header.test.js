import { render, screen, fireEvent } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Header from '../Header';
import '@testing-library/jest-dom';

expect.extend({ toMatchDiffSnapshot });

test('@Header Rocket Navigation', () => {
  const { asFragment } = render(
    <HashRouter>
      <Header />
    </HashRouter>,
  );
  fireEvent.click(screen.getByText(/Rocket/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Rocket/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header Mission Navigation', () => {
  const { asFragment } = render(
    <HashRouter>
      <Header />
    </HashRouter>,
  );
  fireEvent.click(screen.getByText(/Mission/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Mission/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header Profile Navigation', () => {
  const { asFragment } = render(
    <HashRouter>
      <Header />
    </HashRouter>,
  );
  fireEvent.click(screen.getByText(/Profile/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Profile/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
