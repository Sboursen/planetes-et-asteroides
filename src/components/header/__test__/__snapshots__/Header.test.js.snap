// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Header About Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Home Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Rocket Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Header @Header About Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Header @Header Home Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
